### PR TITLE
Fixes bug to allow interface tests to work without TF/PyTorch installed

### PR DESCRIPTION
**Pull request by josh146**
**Context:** A previous PR (#381) added gradient tests to the interfaces. As part of this PR, QNodes were being created at the module level. This was causing exceptions if PyTorch or TensorFlow was not installed, as they were not within scope of the 'skip_if_not_installed' fixtures.

**Description of the Change:** Moves the module level QNodes into the test classes, making them fixtures.

**Benefits:** QNodes will only be created if the corresponding interface is installed.

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: tests/test_tf.py [hunk 1]
```diff
@@ -554,29 +554,31 @@ def circuit_tf(phi, theta):
...
 
-dev = qml.device("default.qubit", wires=2)
-
-
-@qml.qnode(dev, interface="tf")
-def f(x):
-    qml.RX(x, wires=0)
-    return qml.expval(qml.PauliZ(0))
-
+@pytest.mark.usefixtures("skip_if_no_tf_support")
```
Comment (antalszava) [inline tests/test_tf.py]: (A question unrelated to this specific change)

Is there a certain reason why ``tf_support`` is defined in this file (lines 46-56), whereas ``torch_support`` is defined in ``conftest.py``?
```python

@pytest.fixture(scope='module')
def tf_support():
    """Boolean fixture for TensorFlow support"""
    try:
        import tensorflow as tf
        tf_support = True

    except ImportError as e:
        tf_support = False

    return tf_support
```

Comment (antalszava) [reply tests/test_tf.py]: Also, I am wondering about the ``scope`` for these two functions, is there a reason why for ``tf_support`` it's set to ``'module'``, but for ``torch_support`` it's set to ``'session'``?

Comment (josh146) [reply tests/test_tf.py]: These changes have been made in #357, so I won't touch them here to avoid merge conflicts :)
